### PR TITLE
add rpn target assign op

### LINKED ISSUE #10462: Fluid Faster RCNN implementation
### According to [Detectron](https://github.com/facebookresearch/Detectron) the ops need to be implemented for Faster RCNN:
Detectron  |  Paddle Fluid
-----------|----------
[add_rpn_blobs](https://github.com/facebookresearch/Detectron/blob/master/detectron/roi_data/rpn.py#L62:5)|  [generate_anchors_op](https://github.com/facebookresearch/Detectron/blob/e5bb3a8ff0b9caf59c76037726f49465d6b9678b/detectron/roi_data/data_utils.py#L50:5) <br> iou_similarity_op <br> [fg_bg_target_assign_op](https://github.com/facebookresearch/Detectron/blob/e5bb3a8ff0b9caf59c76037726f49465d6b9678b/detectron/roi_data/rpn.py#L195:5) <br> argmax_op <br> box_coder_op
[generate_proposals](https://github.com/facebookresearch/Detectron/blob/master/detectron/ops/generate_proposals.py) | box_coder_op <br> [clip_and_filter_op](https://github.com/facebookresearch/Detectron/blob/0375b05e867eae12be0ca8c420c9efad687689bf/detectron/ops/generate_proposals.py#L152:31) <br> multiclass_nms_op <br>
[generate_proposal_labels](https://github.com/facebookresearch/Detectron/blob/master/detectron/ops/generate_proposal_labels.py) | [add_proposals_op](https://github.com/facebookresearch/Detectron/blob/0375b05e867eae12be0ca8c420c9efad687689bf/detectron/datasets/json_dataset.py#L328:5) <br> [add_gt_proposals_op](https://github.com/facebookresearch/Detectron/blob/a60aa096175d7c1811e0a25d26578cc24489399d/detectron/datasets/roidb.py#L68:5)<br> [target_assign_op](https://github.com/facebookresearch/Detectron/blob/0375b05e867eae12be0ca8c420c9efad687689bf/detectron/datasets/json_dataset.py#L328:5)

### Faster RCNN on Fluid

- **generate_anchors_op**: generate anchors on the fly corresponding to configs
    - Input:
    - Attr:
    - Output: anchor boxes with shape [H, W, M, 4]. H * W * M is total number of anchor box<br>
- **fg_bg_target_assign_op**:
    - Input:
        Input(1): the anchor boxes with shape [M, 4]
        Input(2): the gt boxes with shape [N, 4]
    - Attr:
    - Output: anchor to gt matched mask with shape [M, 1]<br>
- **box_coder_op**:
    - Input: Same
    - Attr:
        Attr(1): add normalize(bool)
    - Output: Same <br>
- **clip_and_fliter_op**: clip and filter proposals corresponding to input image
    - Input: 
        Input(1): decoded anchors with shape [M, 4]
    - Attr:
        Attr(1): image width and height
    - Output: 
        Output(1):cliped and filted proposals with shape [K, 1]<br>
- **add_gt_proposals_op**:
    - Input:
        Input(1): the proposals with shape [M, 4]
        Input(2): the gt boxes with shape [N, 4]
    - Attr:
    - Output: the proposals with shape [M+N, 4] <br> 
- **add_proposals_op**:
    - Input:
    - Attr:
    - Output:<br>
- **target_assign_op**:
    - Input:
    - Attr:
    - Output:<br>

### PR DESCRIPTION
**Pull request by Noplz**
Fix #10462 
Reference [Detectron's Implementation](https://github.com/facebookresearch/Detectron/blob/master/detectron/roi_data/rpn.py)
TBD  c++ and python api doc

### CODE REVIEW

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+namespace operators {
+
+using Tensor = framework::Tensor;
+using LoDTensor = framework::LoDTensor;
+
+class RpnTargetAssignOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+  void InferShape(framework::InferShapeContext* ctx) const override {}
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: Need implementation in `InferShape`.

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    AddAttr<float>("fg_fraction", "").SetDefault(0.25);
+    AddAttr<int>("rpn_batch_size_per_im", "").SetDefault(256);
+    AddAttr<bool>("fix_seed", "").SetDefault(false);
+    AddAttr<int>("seed", "").SetDefault(0);
+    AddOutput("LocationIndex", "");
+    AddOutput("ScoreIndex", "");
+    AddOutput("TargetLabel", "");
+    AddComment(R"DOC(
+)DOC");
+  }
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: Please complete the doc.

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+};
+
+class RpnTargetAssignOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("Overlap", "");
+    AddAttr<float>("rpn_positive_overlap", "").SetDefault(0.7);
+    AddAttr<float>("rpn_negative_overlap", "").SetDefault(0.3);
+    AddAttr<float>("fg_fraction", "").SetDefault(0.25);
+    AddAttr<int>("rpn_batch_size_per_im", "").SetDefault(256);
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: What is the meaning of this attr？

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+                   const float neg_threshold, int* target_label_data,
+                   std::vector<int>* fg_inds, std::vector<int>* bg_inds) const {
+    int fg_offset = fg_inds->size();
+    int bg_offset = bg_inds->size();
+    for (int64_t i = 0; i < row; ++i) {
+      T max_dist = -1;
+      for (int64_t j = 0; j < col; ++j) {
+        T val = dist_data[i * col + j];
+        if (val > max_dist) max_dist = val;
+      }
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: Can use `std::max_element`:

```c++
T* v = dist_data + i * col;
T  max_dist = *std::max_element(v, v + row);
```

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+      for (int64_t j = 0; j < col; ++j) {
+        T val = dist_data[i * col + j];
+        if (val == max_dist) target_label_data[j] = 1;
+      }
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: 这个循环可以上面循环合并吧，找出max_dist的index，给target_label_data赋1即可吧。

**Review by qingqing01 — COMMENTED** [review]
~~Not finish yet.~~

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+                       std::vector<int>* fg_inds, std::vector<int>* bg_inds,
+                       int* target_label_data) const {
+    auto* dist_data = dist.data<T>();
+    int64_t row = dist.dims()[0];
+    int64_t col = dist.dims()[1];
+    int fg_offset = fg_inds->size();
+    int bg_offset = bg_inds->size();
+    std::vector<float> anchor_to_gt_max;
+    // Calculate the max IoU between anchors and gt boxes
+    AnchorToGt(dist_data, row, col, &anchor_to_gt_max);
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: 按列求max，可以采用Eigen::Tensor的maximum操作。

参考： https://github.com/eigenteam/eigen-git-mirror/tree/master/unsupported/Eigen/CXX11/src/Tensor#reduction-dimensions

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+    // Follow the Faster RCNN's implementation
+    ScoreAssign(dist_data, anchor_to_gt_max, row, col, pos_threshold,
+                neg_threshold, target_label_data, fg_inds, bg_inds);
+    // Reservoir Sampling
+    std::uniform_real_distribution<float> uniform(0, 1);
+    if (fg_inds->size() > fg_num) {
+      for (int i = fg_num; i < fg_inds->size(); ++i) {
+        int rng_ind = std::floor(uniform(engine) * fg_num);
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]:
```c++
 std::default_random_engine generator;
 std::uniform_int_distribution<int> distribution(0, i);
```

Reservoir Sampling可以抽出来作为独立的函数吧。

File: python/paddle/fluid/layers/detection.py [hunk 8]
```diff
@@ -44,6 +45,102 @@
...
 
 
+def rpn_target_assign(location,
+                      scores,
+                      anchor_box,
+                      gt_box,
+                      rpn_batch_size_per_im=256,
+                      fg_fraction=0.25,
+                      rpn_positive_overlap=0.7,
+                      rpn_negative_overlap=0.3):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py:55]: 如果这些值不是固定值，用户会经常调，这里就不要设置默认值？ 给用户的接口需要谨慎。

File: python/paddle/fluid/tests/unittests/test_rpn_target_assign_op.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    tgt_lbl[anchors_with_max_overlap] = 1
+    tgt_lbl[anchor_to_gt_max >= rpn_positive_overlap] = 1
+
+    num_fg = int(fg_fraction * rpn_batch_size_per_im)
+    fg_inds = np.where(tgt_lbl == 1)[0]
+    if len(fg_inds) > num_fg:
+        disable_inds = np.random.choice(
+            fg_inds, size=(len(fg_inds) - num_fg), replace=False)
+        tgt_lbl[disable_inds] = -1
+    fg_inds = np.where(tgt_lbl == 1)[0]
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_rpn_target_assign_op.py:39]: 如果这个分支没测到话，最好写个单独的测试，调用 Executor， check一下C++里index的唯一性。

File: python/paddle/fluid/layers/detection.py (hunk 8 above)
Comment (Noplz) [reply python/paddle/fluid/layers/detection.py:55]: 这些值实际用起来的时候，基本都不会去修改的，留给用的话 反而可能不知道填什么合适

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 5 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: 上面改成val >= max_dist看起来是可以合并，不过上面改成max_element之后也没法合并了吧

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 4 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: v + rol ?  done

File: python/paddle/fluid/tests/unittests/test_rpn_target_assign_op.py (hunk 9 above)
Comment (Noplz) [reply python/paddle/fluid/tests/unittests/test_rpn_target_assign_op.py:39]: 这个有例子么

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 1 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 3 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 6 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+        "box for the (anchor, gt box) pair to be a positive example")
+        .SetDefault(0.7);
+    AddAttr<float>(
+        "rpn_negative_overlap",
+        "Maximum overlap allowed between an anchor and ground-truth "
+        "box for the (anchor, gt box) pair to be a negative examples")
+        .SetDefault(0.3);
+    AddAttr<float>("fg_fraction",
+                   "Target fraction of RoI minibatch that "
+                   "is labeled foreground (i.e. class > 0).")
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: (i.e. class > 0) -> (i.e. class > 0, 0-th class is background.)

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+        .SetDefault(false);
+    AddAttr<int>("seed", "RpnTargetAssign random seed.").SetDefault(0);
+    AddOutput(
+        "LocationIndex",
+        "(Tensor), The index of foreground anchors in all rpn anchors, the "
+        "shape of the LocationIndex is [F], F depends on the value of input "
+        "tensor and attributes.");
+    AddOutput("ScoreIndex",
+              "(Tensor), The index of foreground and background anchors in all "
+              "rpn anchors(The rest anchors are ignored). the shape of the "
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: index -> indexes
anchors(The -> anchors (the   注意留空格和大小写。

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+                  "A flag indicating whether to use a fixed seed to generate "
+                  "random mask. NOTE: DO NOT set this flag to true in "
+                  "training. Setting this flag to true is only useful in "
+                  "unittest or for debug that always the same output units "
+                  "will be dropped.")
+        .SetDefault(false);
+    AddAttr<int>("seed", "RpnTargetAssign random seed.").SetDefault(0);
+    AddOutput(
+        "LocationIndex",
+        "(Tensor), The index of foreground anchors in all rpn anchors, the "
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: index -> indexes
rpn -> RPN，所有小写的rpn，都改成大写。

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 13]
```diff
@@ -0,0 +1,261 @@
...
+              "(Tensor), The index of foreground and background anchors in all "
+              "rpn anchors(The rest anchors are ignored). the shape of the "
+              "ScoreIndex is [F + B], F and B depend on the value of input "
+              "tensor and attributes.");
+    AddOutput(
+        "TargetLabel",
+        "(Tensor), The target labels of each anchor with shape [K * M, 1], "
+        "K and M is the same as they are in DistMat,");
+    AddComment(R"DOC(
+)DOC");
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: Please add doc here.

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+        "shape of the LocationIndex is [F], F depends on the value of input "
+        "tensor and attributes.");
+    AddOutput("ScoreIndex",
+              "(Tensor), The index of foreground and background anchors in all "
+              "rpn anchors(The rest anchors are ignored). the shape of the "
+              "ScoreIndex is [F + B], F and B depend on the value of input "
+              "tensor and attributes.");
+    AddOutput(
+        "TargetLabel",
+        "(Tensor), The target labels of each anchor with shape [K * M, 1], "
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: Tensor -> Tensor<int64_t>

The target labels of each anchor -> The target labels of all anchors

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+    }
+    int rpn_batch_size = context.Attr<int>("rpn_batch_size_per_im");
+    float pos_threshold = context.Attr<float>("rpn_positive_overlap");
+    float neg_threshold = context.Attr<float>("rpn_negative_overlap");
+    float fg_fraction = context.Attr<float>("fg_fraction");
+
+    int fg_num = static_cast<int>(rpn_batch_size * fg_fraction);
+
+    int* target_label_data =
+        tgt_lbl->mutable_data<int>({n * col, 1}, context.GetPlace());
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: 用 `int64_t`吧，Fluid里很多loss的计算，label的类型都是`int64_t`。

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 16]
```diff
@@ -0,0 +1,261 @@
...
+    }
+  }
+
+  void ReservoirSampling(const int num, const int offset,
+                         std::minstd_rand engine,
+                         std::vector<int>* inds) const {
+    std::uniform_real_distribution<float> uniform(0, 1);
+    if (inds->size() > num) {
+      for (int i = num; i < inds->size(); ++i) {
+        int rng_ind = std::floor(uniform(engine) * i);
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc:91]: Why not use `std::uniform_int_distribution`, other than using `std::uniform_real_distribution and std::floor`

Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc:91]: 这个生成器 在定义的时候就要确定生成的范围吧，在循环中需要每次都改变生成随机数的范围，如果用int的话 需要每次都定义一个新的，这里不太清楚哪个效率更高一些。

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 10 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 11 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 12 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 13 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 14 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc (hunk 15 above)
Comment (Noplz) [reply paddle/fluid/operators/detection/rpn_target_assign_op.cc]: done

File: python/paddle/fluid/layers/detection.py [hunk 17]
```diff
@@ -44,6 +45,117 @@
...
+def rpn_target_assign(loc,
+                      scores,
+                      anchor_box,
+                      gt_box,
+                      rpn_batch_size_per_im=256,
+                      fg_fraction=0.25,
+                      rpn_positive_overlap=0.7,
+                      rpn_negative_overlap=0.3):
+    """
+    ** Rpn Target Assign Layer for Faster-RCNN **
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: Target Assign Layer for region proposal network (RPN) in Faster-RCNN detection.

File: python/paddle/fluid/layers/detection.py [hunk 18]
```diff
@@ -44,6 +45,117 @@
...
+
+    This layer can be, for given the IoU between anchors and ground truth boxes,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: IoU -> Intersection-over-Union (IoU) overlap

File: python/paddle/fluid/layers/detection.py [hunk 19]
```diff
@@ -44,6 +45,117 @@
...
+    to assign classification and regression targets to each anchors.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: each anchors -> each anchor, these target labels are used for train RPN. 

The classification targets is a binary class label (of being an object or not).  Following the paper of Faster-RCNN,  the positive labels are two kinds of anchors:  (i) the anchor/anchors with the highest IoU overlap with a ground-truth box, or (ii) an anchor that has an IoU overlap higher than `rpn_positive_overlap`(0.7) with any ground-truth box.  Note that a single ground-truth box may assign positive labels to multiple anchors.  A non-positive anchor is when its IoU ratio is lower than `rpn_negative_overlap` (0.3) for all ground-truth boxes.  Anchors that are neither positive nor negative do not contribute to the training objective.

The regression targets are the encoded ground-truth boxes associated with the positive anchors.

File: python/paddle/fluid/layers/detection.py [hunk 20]
```diff
@@ -44,6 +45,117 @@
...
+    target_bbox = helper.create_tmp_variable(dtype=anchor_box.dtype)
+    helper.append_op(
+        type="box_coder",
+        inputs={
+            'PriorBox': anchor_box,
+            'TargetBox': gt_box,
+        },
+        outputs={'OutputBox': target_bbox},
+        attrs={'code_type': 'encode_center_size',
+               'box_normalized': False})
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: 这里可以直接调用 box_coder 的接口那，而不用 append_op这种吧：
```
    target_bbox = box_coder(
        prior_box=anchor_box,
        target_box=gt_box,
        code_type='encode_center_size',
        box_normalized=False)
```

File: python/paddle/fluid/layers/detection.py [hunk 21]
```diff
@@ -44,6 +45,117 @@
...
+    Args:
+        loc(Variable): A 3-D Tensor with shape [N, M, 4] represents the
+            predicted locations of M bounding bboxes. N is the batch size,
+            and each bounding box has four coordinate values and the layout
+            is [xmin, ymin, xmax, ymax].
+        scores(Variable): A 3-D Tensor with shape [N, M, C] represents the
+            predicted confidence predictions. N is the batch size, C is the
+            class number, M is number of bounding boxes. For each category
+            there are total M scores which corresponding M bounding boxes.
+        anchor_box(Variable): A 2-D Tensor with shape [M, 4] holds M boxes,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py:83]: 注意，每行不超过80个字符。

File: python/paddle/fluid/layers/detection.py [hunk 22]
```diff
@@ -44,6 +45,117 @@
...
+        rpn_positive_overlap(float): Minimum overlap required between an anchor and ground-truth box for the (anchor, gt box) pair to be a positive example.
+        rpn_negative_overlap(float): Maximum overlap allowed between an anchor and ground-truth box for the (anchor, gt box) pair to be a negative examples.
+
+    Returns:
+        tuple: 
+               A tuple(predicted_scores, predicted_location, target_label, target_bbox) is returned. 
+               The predicted_scores and predicted_location is the predicted result of the RPN. The target_label and target_bbox is the ground truth, respectively.
+               The predicted_location is a 2D Tensor with shape [F, 4], and the shape of target_bbox is same as the shape of the predicted_location, F is the
+               number of the foreground anchors. The predicted_scores is a 2D Tensor with shape [F + B, 1], and the shape of target_label
+               is same as the shape of the predicted_scores, B is the number of the background anchors, the F and B is depends on the input of this operator.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: 同上，每行不超过80个字符。

File: paddle/fluid/operators/detection/rpn_target_assign_op.cc [hunk 23]
```diff
@@ -0,0 +1,285 @@
...
+is exactly same as the foreground anchor indexes since we can not assign regression target to the
+background anchors.
+
+This operator select foreground and background anchors by performing the following steps:
+
+1. We can obtain each anchor's matched ground truth bbox according to the IoU(Anchor, gtBBox)
+
+2. For each anchor:
+
+If Max(IoU(Anchor, GTBBoxes)) > rpn_positive_overlap:
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/rpn_target_assign_op.cc]: 还有一种anchor会标记为正样本吧：

the anchor/anchors with the highest IoU overlap with a ground-truth box

见Python文档里的review。

**Review by qingqing01 — COMMENTED** [review]
Mainly for documentation.

**Review by qingqing01 — APPROVED** [review]

File: python/paddle/fluid/layers/detection.py (hunk 17 above)
Comment (Noplz) [reply python/paddle/fluid/layers/detection.py]: done

File: python/paddle/fluid/layers/detection.py (hunk 18 above)
Comment (Noplz) [reply python/paddle/fluid/layers/detection.py]: done

File: python/paddle/fluid/layers/detection.py (hunk 19 above)
Comment (Noplz) [reply python/paddle/fluid/layers/detection.py]: done

File: python/paddle/fluid/layers/detection.py (hunk 21 above)
Comment (Noplz) [reply python/paddle/fluid/layers/detection.py:83]: done

File: python/paddle/fluid/layers/detection.py (hunk 22 above)
Comment (Noplz) [reply python/paddle/fluid/layers/detection.py]: done

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**